PerfNext/app/apis/BenchEngine/parser.js

```diff
@@ -41,24 +41,44 @@ module.exports = function(app) {
     */
 	app.post('/api/benchengine/submit', function(req, res) {
 		var raw_definition = req.rawBody; //Raw XML to be parsed
-
-		parseDefinition(raw_definition, function(generatedOutput){
-
-			var setupScript = parseSetupScript(raw_definition, function(setupScript){
-
+		
+		// Split the received xml file to separate test and baseline scripts
+		var split_definition = raw_definition.split("<?xml");
+		
+		// Add in missing xml tag from previous split
+		var test_raw_definition = "<?xml" + split_definition[1];
+		
+			
+		// If interleave baseline unticked
+		if (split_definition.length < 3) {
+			var baseline_raw_defintion = null;	
+		} else {
+			var baseline_raw_definition = "<?xml" + split_definition[2];
+		}
+		parseDefinition(test_raw_definition, function(generatedOutput){
+			
+			var setupScript = parseSetupScript(test_raw_definition, function(setupScript){
+				
+				parseDefinition(baseline_raw_definition, function(generatedOutputbase){
+					var setupScriptbase = parseSetupScript(baseline_raw_definition, function(setupScriptbase){
 				//console.log('parser.js: setupScript: '+setupScript);
 				
 				var setupMachine=req.query.machine;
 				console.log('parser.js: setupMachine: '+setupMachine);
 				
+				// Parse iteration number from benchmark script
+				var iteration_ind = generatedOutput.indexOf("iterations = ");
+				var iteration_num = generatedOutput[iteration_ind + "iterations = ".length]
+				console.log("Total Iterations = " + iteration_num);
+				
 				var jenkinsAPI =  global.APP_DATA.jenkins_base + "/buildWithParameters?token=xYz@123&setupScript="+encodeURIComponent(setupScript)+'&setupMachine='+encodeURIComponent(setupMachine)+'&benchmarkScript='+encodeURIComponent(generatedOutput)+'&benchmarkMachine='+encodeURIComponent(req.query.machine);  
 
 				console.log(`the selected machine is ${req.query.machine}`);
 
 				//console.log('parser.js: Calling jenkinsAPI: '+jenkinsAPI);
 				
 				var jenkinsURL =  global.APP_DATA.jenkins_base + "/buildWithParameters"
-				var postBody = "token=xYz@123&setupScript="+encodeURIComponent(setupScript)+'&setupMachine='+encodeURIComponent(setupMachine)+'&benchmarkScript='+encodeURIComponent(generatedOutput)+'&benchmarkMachine='+encodeURIComponent(req.query.machine);  
+				var postBody = "token=xYz@123&setupScripttest="+encodeURIComponent(setupScript)+'&setupMachine='+encodeURIComponent(setupMachine)+'&benchmarkScripttest='+encodeURIComponent(generatedOutput)+'&benchmarkMachine='+encodeURIComponent(req.query.machine)+'&benchmarkScriptbaseline='+encodeURIComponent(generatedOutputbase)+'&setupScriptbaseline='+encodeURIComponent(setupScriptbase)+'&Iterations='+encodeURIComponent(iteration_num);  
```

Comment:
When you make a common function, you can move this outside.

Reply:
Just to be consistent, maybe we can use `generatedOutputBaseline` instead of `generatedOutputbase`.